### PR TITLE
metricbeat/module/postgresql: Improve healthcheck for Postgres

### PR DESCRIPTION
**Pull request by shmsr**
## What does this PR do?

Improve healthcheck for Postgres to reduce the flakiness in integration tests.

## Why is it important?

Flaky tests lead to instability of CI and could lead to unnecessary blocking of PRs and unreliable tests. In my opinion, `start_period` and reason mentioned in https://github.com/peter-evans/docker-compose-healthcheck/issues/16#issuecomment-1420377736 is something that could help us as this improves healthcheck.

Also bumped the fallback version for the image to the latest minor of the same major release to avoid any issues. Currently, my latest major is 15 and currently, I've set the fallback version to 13.11 so that it doesn't change the major.

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.
- [x] I have made my commit title and message explanatory about the purpose and the reason of the change

## How to test this PR locally

As the flaky tests are not reproducible, we'll do a soak testing i.e., soak it for some time to see if it really reduced the flakiness of integration tests or not.

## Related issues

- Relates https://github.com/elastic/obs-infraobs-team/issues/1150
- Closes https://github.com/elastic/beats/issues/35995#issuecomment-1657704361
- Relates https://github.com/elastic/obs-infraobs-team/issues/1150#issuecomment-1657968550
- Relates https://github.com/peter-evans/docker-compose-healthcheck/issues/16#issuecomment-1420377736
- Relates https://github.com/docker-library/postgres/issues/146#issuecomment-872486465
- Relates https://github.com/moby/moby/pull/28679

### CODE REVIEW

**Comment by shmsr** [comment]
Keeping this PR in draft for some time to do some testing myself. But it is ready for review.

**Comment by shmsr** [comment]
^^ Will retrigger a few times in order to test if Postgres tests are failing or not.

File: metricbeat/module/postgresql/docker-compose.yml [hunk 1]
```diff
@@ -2,17 +2,18 @@ version: '2.3'
...
+        POSTGRESQL_VERSION: ${POSTGRESQL_VERSION:-13.11}
     environment:
       POSTGRES_PASSWORD: postgres
     ports:
       - 5432
     healthcheck:
-      test: ["CMD-SHELL", "pg_isready -U postgres -p 5432"]
-      interval: 5s
-      timeout: 5s
+      test: ['CMD-SHELL', 'psql -h localhost -p 5432 -U postgres -c select 1 -d postgres']
```
Comment (agithomas) [inline metricbeat/module/postgresql/docker-compose.yml:15]: `select 1` behaviour is not very deterministic, does not return an output. 

Do you want to consider running `selection version()` instead?

What advantage `select 1` has over `pg_is_ready`, when `pg_is_ready` indicate the database is ready to accept the connection

Comment (shmsr) [reply metricbeat/module/postgresql/docker-compose.yml:15]: If you see the referenced links in PR description you can find more info but I'll put a short summary here:

- So there are two phases in which Postgres container starts up twice, one is initialization and then for real to which our testing suite is supposed to connect. As there was no `start_period` before and the usage of `pg_isready` resulted in HEALTHCHECK pass which might not be the case. Refer: https://github.com/Silicon-Ally/testpgx/blob/d198f78f1d1f18933750826f3f07f209c87b4a0b/testpgx.go#L184

- Also see: https://stackoverflow.com/a/74095912/5821408. With `pg_isready` we cannot even check with `pg_isready` if a db exists but with `psql` we can.

- Read: https://github.com/peter-evans/docker-compose-healthcheck/issues/16#issuecomment-1420377736 (there's also a note at the end). So I've set the `timeout`, `interval`, `start_period` properly to do the HEALTHCHECK more reliably now and psql now actually selects 1 row from an existing table (`select 1`) which confirms it is in a ready state and not in another state (like initialization state).

You can refer to the linked issues for more details.

Comment (shmsr) [reply metricbeat/module/postgresql/docker-compose.yml:15]: > select 1 behaviour is not very deterministic, does not return an output.

We are depending on the exit code of `psql` and not `select 1` as if `select 1` fails then psql exits with a non-zero exit code. It is fine to not have rows for a particular db but it is ensuring that it is able to connect to the same.

Comment (agithomas) [reply metricbeat/module/postgresql/docker-compose.yml:15]: Thanks for taking the time for explaining in a detailed manner.

Comment (shmsr) [reply metricbeat/module/postgresql/docker-compose.yml:15]: Thanks for reviewing 🚀

**Pull request merged**